crates/yttrium/src/chain_abstraction/pulse.rs

```diff
@@ -0,0 +1,137 @@
+            .map(|url| url.origin().ascii_serialization()),
+        bundle_id: pulse_metadata.bundle_id.clone(),
+        timestamp,
+        props,
+    };
+    debug!("pulse analytics: {analytics:?}");
+
+    let query = Query {
+        project_id,
+        sdk_type: SDK_TYPE,
+        sdk_version: pulse_metadata.sdk_version.clone(),
+        sdk_platform: pulse_metadata.sdk_platform.clone(),
+    };
+
+    // println!(
+    //     "url: {}",
+    //     http_client.post(PULSE_ENDPOINT).query(&query).build().unwrap().url()
+    // );
+
+    let fut = async move {
+        let result = http_client
+            .post(PULSE_ENDPOINT)
+            .query(&query)
+            .json(&analytics)
+            .send()
+            .await;
+        match result {
+            Ok(response) => {
+                let status = response.status();
+                if status.is_success() {
+                    debug!("successfully sent execute() analytics");
+                } else {
+                    match response.text().await {
+                        Ok(text) => {
+                            warn!("execute() analytics request failed: {status} {text}");
+                        }
+                        Err(e) => {
+                            warn!("execute() analytics request failed: {e}");
+                        }
+                    }
+                }
+            }
+            Err(e) => {
+                warn!("execute() analytics request failed: {e}");
+            }
+        }
+    };
+
+    #[cfg(not(test))]
+    super::spawn::spawn(fut);
+
+    #[cfg(test)]
+    std::thread::spawn(move || {
+        tokio::runtime::Builder::new_current_thread()
+            .enable_all()
+            .build()
+            .unwrap()
+            .block_on(fut)
+    })
+    .join()
+    .unwrap();
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Event {
+    pub event_id: Uuid,
+    pub url: Option<String>,
+    pub domain: Option<String>,
+    pub bundle_id: Option<String>,
+    #[serde(with = "duration_millis")]
+    pub timestamp: Duration,
+    pub props: ExecuteAnalytics,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Query {
+    pub project_id: ProjectId,
+    #[serde(rename = "st")]
+    pub sdk_type: &'static str,
+    #[serde(rename = "sv")]
+    pub sdk_version: String,
+    #[serde(rename = "sp")]
+    pub sdk_platform: String,
+}
+
+const SDK_TYPE: &str = "wkca";
+
+#[derive(Clone, Serialize, Deserialize)]
+#[cfg_attr(feature = "uniffi", derive(uniffi::Record))]
+#[cfg_attr(
+    feature = "wasm",
+    derive(tsify_next::Tsify),
+    tsify(into_wasm_abi, from_wasm_abi)
+)]
+pub struct PulseMetadata {
+    // web
+    pub url: Option<Url>,
+    // iOS
+    pub bundle_id: Option<String>,
+    // Android
+    // FIXME this param is not used yet
+    pub package_name: Option<String>,
+    pub sdk_version: String,
+    pub sdk_platform: String,
```

Comment:
whats this?

Reply:
`sdk_platform` is a Pulse API parameter, It accepts `web`, `mobile`, or `desktop`